src/m365/pp/commands/tenant/tenant-settings-set.ts

```diff
@@ -0,0 +1,257 @@
+          typeof args.options.disablePortalsCreationByNonAdminusers === 'undefined' &&
+          typeof args.options.disableSurveyFeedback === 'undefined' &&
+          typeof args.options.disableTrialEnvironmentCreationByNonAdminusers === 'undefined' &&
+          typeof args.options.disableCapacityAllocationByEnvironmentAdmins === 'undefined' &&
+          typeof args.options.disableSupportTicketsVisibleByAllUsers === 'undefined' &&
+          typeof args.options.disableDocsSearch === 'undefined' &&
+          typeof args.options.disableCommunitySearch === 'undefined' &&
+          typeof args.options.disableBingVideoSearch === 'undefined' &&
+          !args.options.shareWithColleaguesUserLimit &&
+          typeof args.options.disableShareWithEveryone === 'undefined' &&
+          typeof args.options.enableGuestsToMake === 'undefined' &&
+          typeof args.options.disableAdminDigest === 'undefined' &&
+          typeof args.options.disableDeveloperEnvironmentCreationByNonAdminUsers === 'undefined' &&
+          typeof args.options.disableBillingPolicyCreationByNonAdminUsers === 'undefined' &&
+          typeof args.options.disableChampionsInvitationReachout === 'undefined' &&
+          typeof args.options.disableSkillsMatchInvitationReachout === 'undefined') {
+          return 'Specify at least one property to update';
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    const data = {
+      walkMeOptOut: args.options.walkMeOptOut,
+      disableNPSCommentsReachout: args.options.disableNPSCommentsReachout,
+      disableNewsletterSendout: args.options.disableNewsletterSendout,
+      disableEnvironmentCreationByNonAdminUsers: args.options.disableEnvironmentCreationByNonAdminUsers,
+      disablePortalsCreationByNonAdminUsers: args.options.disablePortalsCreationByNonAdminUsers,
+      disableSurveyFeedback: args.options.disableSurveyFeedback,
+      disableTrialEnvironmentCreationByNonAdminUsers: args.options.disableTrialEnvironmentCreationByNonAdminUsers,
+      disableCapacityAllocationByEnvironmentAdmins: args.options.disableCapacityAllocationByEnvironmentAdmins,
+      disableSupportTicketsVisibleByAllUsers: args.options.disableSupportTicketsVisibleByAllUsers,
+      powerPlatform: {
+        search: {
+          disableDocsSearch: args.options.disableDocsSearch,
+          disableCommunitySearch: args.options.disableCommunitySearch,
+          disableBingVideoSearch: args.options.disableBingVideoSearch
+        },
+        teamsIntegration: {
+          shareWithColleaguesUserLimit: args.options.shareWithColleaguesUserLimit
+        },
+        powerApps: {
+          disableShareWithEveryone: args.options.disableShareWithEveryone,
+          enableGuestsToMake: args.options.enableGuestsToMake,
+          disableMembersIndicator: args.options.disableMembersIndicator
```

Comment:
Seems like I forgot to spec out this option, it's not in the docs or in the options list. Let's try to add it.